src/pages/index.js

```diff
@@ -1,105 +1,277 @@
+const avatarEditFormPopup = new PopupWithForm({
+  popupSelector: "#avatar-edit-modal",
+  handleFormSubmit: handleAvatarEditSubmit,
+});
 
-profileEditButton.addEventListener("click", function () {
-  const { name, description } = profileUserInfo.getUserInfo();
-  profileTitleInput.value = name;
-  profileDescriptionInput.value = description;
-  profileEditFormPopup.open();
+const avatarUserInfo = new UserInfo({
+  avatarEl: ".profile__image",
 });
 
+const cardSection = new Section(
+  {renderer: renderCard},
+  ".cards__list"
+);
+
+const newItemPopup = new PopupWithForm({
+  popupSelector: "#new-item-modal",
+  handleFormSubmit: handleNewItemSubmit,
+});
+
+const deleteCardPopup = new PopupWithFormDelete({
+  popupSelector: "#delete-card-modal",
+});
+
+const previewImagePopup = new PopupWithImage("#preview-image-modal");
+
+/* -------------------------------------------------------------------------- */
+/*                                  API Calls                                 */
+/* -------------------------------------------------------------------------- */
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    profileUserInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+    })
+    profileUserInfo.setAvatarInfo({
+      link: userData.avatar,
+    })
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+/* -------------------------------------------------------------------------- */
+/*                               Event Handlers                               */
+/* -------------------------------------------------------------------------- */
+
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      popupInstance.close()
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
+}
+
 function handleProfileEditSubmit(userData) {
   const name = userData.title;
   const description = userData.description;
-  profileUserInfo.setUserInfo({ name, description });
-  profileEditFormPopup.close();
+
+  function makeRequest() {
+    return api.updateUserInfo(name, description)
+    .then((newUserData) => {
+      profileUserInfo.setUserInfo({
+        name: newUserData.name,
+        description: newUserData.about,
+        id: newUserData._id, //capture user id
+      });
+      profileEditForm.reset();
+    })
+  }
+  handleSubmit(makeRequest, profileEditFormPopup);
 }
 
-/* ------------------------------ Adding cards ------------------------------ */
+function handleAvatarEditSubmit(inputValue) {
+  const link = inputValue.link;
+  function makeRequest() {
+    return api.updateProfileAvatar(link)
+    .then((newUserData) => {
+      formValidators["avatar-edit-form"].resetValidation();
```

Comment:
`resetValidation` is not needed here


Method `resetValidation` is for clearing possible validation errors, when you open the **profile** form, because you insert **valid** data into its inputs.